### PR TITLE
Add net10.0 known framework references

### PR DESCRIPTION
**Pull request by v-wuzhai**

### CODE REVIEW

**Comment by v-wuzhai** [comment]
@dotnet/source-build-internal Could you take a look at the failures here?

```
 Failed Microsoft.DotNet.SourceBuild.SmokeTests.WebScenarioTests.VerifyScenario(scenario: TestScenario { Commands = PublishSelfContained, Language = CSharp, NoHttps = False, ScenarioName = "WebScenarioTests", Template = WebApp, ··· }) [9 s]
  Error Message:
   Runtime packs were retrieved from NuGet instead of the SDK
  Stack Trace:
     at Microsoft.DotNet.SourceBuild.SmokeTests.WebScenarioTests.VerifyRuntimePacksForSelfContained(String projectPath) in /vmr/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs:line 55
   at Microsoft.DotNet.SourceBuild.SmokeTests.TestScenario.Execute(DotNetHelper dotNetHelper) in /vmr/test/Microsoft.DotNet.SourceBuild.SmokeTests/TestScenario.cs:line 75
   at Microsoft.DotNet.SourceBuild.SmokeTests.WebScenarioTests.VerifyScenario(TestScenario scenario) in /vmr/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs:line 25
   at InvokeStub_WebScenarioTests.VerifyScenario(Object, Span`1)
   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)

```

**Comment by ellahathaway** [comment]
Looking - this test verifies that the runtime packs are not being pulled from NuGet. When they are, they end up in the cache and the test fails. Based on this, `dotnet publish --self-contained` is using NuGet to pull the runtime packs instead of the sdk.

Based on, https://github.com/dotnet/installer/pull/16499 I think that @tmds may be the one who introduced some of the functionality into source-build that this test verifies. I also see https://github.com/dotnet/source-build/issues/1215 as a related issue.

@MichaelSimons @tmds  - do either of you have thoughts on what may be causing this test to fail?

**Comment by MichaelSimons** [comment]
I think this test is going to need to be disabled through the net10.0 transition period.  A new issue should be opened to track the re-enablement.

**Review by MichaelSimons — APPROVED** [review]

File: src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs [hunk 1]
```diff
@@ -37,7 +37,8 @@ private static IEnumerable<TestScenario> GetScenarios()
 
         yield return new(nameof(WebScenarioTests), DotNetLanguage.CSharp, DotNetTemplate.Razor,         DotNetActions.Build | DotNetActions.Run | DotNetActions.Publish);
         yield return new(nameof(WebScenarioTests), DotNetLanguage.CSharp, DotNetTemplate.BlazorWasm,    DotNetActions.Build | DotNetActions.Run | DotNetActions.Publish);
-        yield return new(nameof(WebScenarioTests), DotNetLanguage.CSharp, DotNetTemplate.WebApp,        DotNetActions.PublishSelfContained, VerifyRuntimePacksForSelfContained);
+        // Disabled due to .NET 10.0 transition. See https://github.com/dotnet/sdk/pull/42969
```
Comment (akoeplinger) [inline src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs:40]: please file a tracking issue for this

**Review by akoeplinger — APPROVED** [review]

**Comment by tmds** [comment]
> Runtime packs were retrieved from NuGet instead of the SDK

The error indicates that the SDK isn't finding a runtime pack that should be produced by the build. I don't know why this would stop working during bringup. I'm ok with disabling the test for some time, and perhaps it will resolve itself.

File: src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs (hunk 1 above)
Comment (ellahathaway) [reply src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/WebScenarioTests.cs:40]: https://github.com/dotnet/source-build/issues/4578

**Pull request merged**